src/ucp/rndv/proto_rndv.inl

```diff
@@ -45,13 +45,16 @@ ucp_proto_rndv_rts_request_init(ucp_request_t *req)
         return status;
     }
 
-    status = ucp_datatype_iter_mem_reg(ep->worker->context,
-                                       &req->send.state.dt_iter, rpriv->md_map,
-                                       UCT_MD_MEM_ACCESS_RMA |
-                                               UCT_MD_MEM_FLAG_HIDE_ERRORS,
-                                       UCP_DT_MASK_ALL);
-    if (status != UCS_OK) {
-        return status;
+    if (rpriv->md_map != 0) {
```

Comment:
is this if really needed?

Reply:
yes, otherwise we will register mem for IOVs, even if it is not really needed.

Reply:
maybe it should be in ucp_datatype_iter_mem_reg?